app/src/main/java/com/peshale/nmedia/vmodel/PostViewModel.kt

```diff
@@ -2,60 +2,139 @@ package com.peshale.nmedia.vmodel
 
 import android.app.Application
 import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import com.peshale.nmedia.db.AppDb
 import com.peshale.nmedia.dto.Post
+import com.peshale.nmedia.model.FeedModel
 import com.peshale.nmedia.repository.PostRepository
-import com.peshale.nmedia.repository.PostRepositorySQLiteImpl
+import com.peshale.nmedia.repository.PostRepositoryImpl
 import com.peshale.nmedia.utils.AndroidUtils
+import com.peshale.nmedia.utils.SingleLiveEvent
+import java.io.IOException
+import java.util.concurrent.Executors
+import kotlin.concurrent.thread
 
 private val empty = Post(
     id = 0,
     content = "",
     author = "",
+    authorAvatar = "",
     likedByMe = false,
     published = AndroidUtils.addLocalDataTime(),
-    likes = 0,
-    shares = 0,
-    views = 0,
-    video = ""
+    likes = 0
 )
 
-
 class PostViewModel(application: Application) : AndroidViewModel(application) {
-    private val repository: PostRepository = PostRepositorySQLiteImpl(AppDb.getInstance(application).postDao)
-    val data = repository.getAll()
+    private val repository: PostRepository = PostRepositoryImpl()
+    private val _data = MutableLiveData(FeedModel())
+    private val executorService = Executors.newFixedThreadPool(64)
+
+    val data: LiveData<FeedModel>
+        get() = _data
     val edited = MutableLiveData(empty)
 
-    fun addPost() {
+    private val _postAdded = SingleLiveEvent<Unit>()
+    val postAdded: LiveData<Unit>
+        get() = _postAdded
+
+    var actualLikedByMe: MutableMap<Long, Boolean> = emptyMap<Long, Boolean>().toMutableMap()
+
+    init {
+        loadPosts()
+    }
+
+    fun loadPosts() {
+        _data.postValue(FeedModel(loading = true))
+        repository.getAllAsync(object : PostRepository.GetAllCallback {
+            override fun onSuccess(posts: List<Post>) {
+                posts.map {
+                    if (actualLikedByMe.containsKey(it.id)) {
+                        actualLikedByMe[it.id]?.let { it_ -> it.copy(likedByMe = it_) }
+                    } else {
+                        it
+                    }
+                }
+                _data.postValue(FeedModel(posts = posts, empty = posts.isEmpty()))
+            }
+
+            override fun onError(e: Exception) {
+                _data.postValue(FeedModel(error = true))
+            }
+        })
+    }
+
+    fun postCreation() {
+        edited.value?.let {
```

Comment:
```suggestion
        edited.value?.let { post ->
```
Тогда следующая строка кода не понадобится